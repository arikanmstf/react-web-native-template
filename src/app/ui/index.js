export { default as Text } from './Text/Web';
export { default as Button } from './Button/Web';
export { default as View } from './View/Web';
