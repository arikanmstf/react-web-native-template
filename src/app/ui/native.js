export { default as Text } from './Text/Native';
export { default as Button } from './Button/Native';
export { default as View } from './View/Native';
export { default as Image } from './Image/Native';
