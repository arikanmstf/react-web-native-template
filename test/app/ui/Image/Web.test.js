import React from 'react';
import { shallow } from 'enzyme';
import { Image } from 'ui';

describe('src/app/ui/Image/Web.js', () => {
  const props = {
    source: {
      uri: 'dummyUri',
    },
  };

  it('Render: props: { source: { uri } }', () => {
    const wrapper = shallow(<Image {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render: props: { alt }', () => {
    const newProps = {
      ...props,
      alt: 'dummyAlt',
    };

    const wrapper = shallow(<Image {...newProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render: props: { style }', () => {
    const newProps = {
      ...props,
      style: { width: '100%' },
    };

    const wrapper = shallow(<Image {...newProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
