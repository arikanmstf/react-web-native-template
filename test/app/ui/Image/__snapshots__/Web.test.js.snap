// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app/ui/Image/Web.js Render: props: { alt } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    alt="dummyAlt"
    source={
      Object {
        "uri": "dummyUri",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "dummyAlt",
      "src": "dummyUri",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "dummyAlt",
        "src": "dummyUri",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`src/app/ui/Image/Web.js Render: props: { source: { uri } } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    source={
      Object {
        "uri": "dummyUri",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "image",
      "src": "dummyUri",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "image",
        "src": "dummyUri",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`src/app/ui/Image/Web.js Render: props: { style } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    source={
      Object {
        "uri": "dummyUri",
      }
    }
    style={
      Object {
        "width": "100%",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "image",
      "src": "dummyUri",
      "style": Object {
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "image",
        "src": "dummyUri",
        "style": Object {
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
